src/neuroconv/datainterfaces/behavior/medpc/medpcdatainterface.py

```diff
@@ -0,0 +1,254 @@
+            to an info dictionary with the names of the variables in the metadata and whether or not they are arrays.
+            ex. {"A": {"name": "left_nose_poke_times", "is_array": True}}
+        """
+        original_timestamps_dict = self.get_original_timestamps(medpc_name_to_info_dict=medpc_name_to_info_dict)
+        aligned_timestamps_dict = {}
+        for name, original_timestamps in original_timestamps_dict.items():
+            aligned_timestamps_dict[name] = original_timestamps + aligned_starting_time
+        self.set_aligned_timestamps(aligned_timestamps_dict=aligned_timestamps_dict)
+
+    def add_to_nwbfile(
+        self,
+        nwbfile: NWBFile,
+        metadata: dict,
+        medpc_name_to_info_dict: Optional[dict] = None,
+        module_name: Optional[str] = None,
+        module_description: Optional[str] = None,
+        events: Optional[list[dict]] = None,
+        interval_series: Optional[list[dict]] = None,
+    ) -> None:
+        ndx_events = get_package(package_name="ndx_events", installation_instructions="pip install ndx-events")
+        medpc_name_to_info_dict = metadata["MedPC"].get("medpc_name_to_info_dict", medpc_name_to_info_dict)
+        info_name_to_medpc_name = {
+            info_dict["name"]: medpc_name for medpc_name, info_dict in medpc_name_to_info_dict.items()
+        }
+        for name in self.source_data["aligned_timestamp_names"]:
+            medpc_name = info_name_to_medpc_name[name]
+            medpc_name_to_info_dict.pop(medpc_name)
+        session_dict = read_medpc_file(
+            file_path=self.source_data["file_path"],
+            medpc_name_to_info_dict=medpc_name_to_info_dict,
+            session_conditions=self.source_data["session_conditions"],
+            start_variable=self.source_data["start_variable"],
+        )
+        aligned_timestamps_dict = self.get_timestamps()
+        for name, aligned_timestamps in aligned_timestamps_dict.items():
+            session_dict[name] = aligned_timestamps
+
+        # Add behavior data to nwbfile
+        module_name = metadata["MedPC"].get("module_name", module_name)
+        module_description = metadata["MedPC"].get("module_description", module_description)
+        behavior_module = nwb_helpers.get_module(
+            nwbfile=nwbfile,
+            name=module_name,
+            description=module_description,
+        )
+
+        event_dicts = metadata["MedPC"].get("Events", events)
+        for event_dict in event_dicts:
+            name = event_dict["name"]
+            description = event_dict["description"]
+            event_data = session_dict[name]
+            if len(event_data) > 0:
+                event = ndx_events.Events(
+                    name=name,
+                    description=description,
+                    timestamps=event_data,
+                )
+                behavior_module.add(event)
+        interval_dicts = metadata["MedPC"].get("IntervalSeries", interval_series)
```

Comment:
Similar to the field above, I would recommend choosing one or the other (metadata or conversion options) for how to fetch/specify this data to avoid confusion about which to use